src/backend/distributed/commands/database.c

```diff
@@ -688,6 +695,45 @@ PreprocessDropDatabaseStmt(Node *node, const char *queryString,
 }
 
 
+List *
+PreprocessAlterDatabaseOwnerStmt(Node *node, const char *queryString,
+								 ProcessUtilityContext processUtilityContext)
+{
+	const DistributeObjectOps *ops = GetDistributeObjectOps(node);
+	Assert(ops != NULL);
+
+	if (ops->featureFlag && *ops->featureFlag == false)
+	{
+		/* not propagating when a configured feature flag is turned off by the user */
+		return NIL;
+	}
+
+	List *addresses = GetObjectAddressListFromParseTree(node, false, false);
+
+	/*  the code-path only supports a single object */
+	Assert(list_length(addresses) == 1);
+
+	if (!ShouldPropagateAnyObject(addresses))
+	{
+		return NIL;
+	}
+	EnsurePropagationToCoordinator();
+
+	AlterOwnerStmt *stmt = (AlterOwnerStmt *) node;
+	EnsureSequentialMode(stmt->objectType);
+	QualifyTreeNode(node);
+	char *databaseName = strVal((String *) stmt->object);
+	SerializeDistributedDDLsOnObjectClassObject(OCLASS_DATABASE, databaseName);
```

Comment:
let's move this to right after EnsurePropagationToCoordinator() because we want to ensure that we acquire this lock before Postgres does anything

Reply:
Done